### ISSUE TITLE
Rails  integration installer broke on Windows?

### ISSUE THREAD

**Issue by davecan** [Compass/compass#67]
Following the screencast on the "Ruby on Rails Integration" page, I typed in the following command:

```
> rails -m http://www.compass-style.org/rails/installer compass_test
```

I did this twice, first typing "960" for the plugin type and accepting all other defaults, and the second time accepting all defaults. Each time it spat out the following at the end of processing and quit:

```
     gem  haml
     gem  chriseppstein-compass
    rake  gems:install GEM=haml
          D:/Dev/Languages/ruby/bin/rails: No such file or directory - sudo rake gems:install GEM=haml RAILS_ENV=development
    rake  gems:install GEM=chriseppstein-compass
          D:/Dev/Languages/ruby/bin/rails: No such file or directory - sudo rake gems:install GEM=chriseppstein-compass RAILS_ENV=development
    rake  gems:unpack GEM=chriseppstein-compass
          D:/Dev/Languages/ruby/lib/ruby/gems/1.8/gems/activesupport-2.3.4/lib/active_support/core_ext/kernel/agnostics.rb:7:in ``': Exec format error - rake gems:unpack GEM=chriseppstein-compass RAILS_ENV=development
```

Both times I had no "app/stylesheets" directory. It appears the installer bailed out before reaching that point. I was able to copy out the rake commands from the above output and manually run each one, resulting in a gem placed in "vendor/gem", but nothing else.

I am running on Windows XP. I notice the error failed on command "sudo", and I do not have a sudo command. I am using the excellent unxutils package from SourceForge, but there is no need for a sudo command. I did download and try sudowin from SourceForge, but it wanted a password and I don't use one for this account. Finally, I tried creating a batch file named "sudo.bat" that would simply take anything after the "sudo" command and simply execute it in the shell, but that still resulted in the above error messages and bailout.

Thanks for looking at this. Any help appreciated.

**Comment by perezd** [comment]
Hey Dave,

I don't have a fully setup Win/Rails environment to test this out on, so I may need your debugging help. Can you try following the examples on this site?

http://blog.anlek.com/2009/04/rails-templates/

My best guess would be that we need to override the rake method (as they do) to support win32, this is actually a problem with the Rails templating system :(

Sorry! Let me know how that goes, and maybe push a fix?

**Comment by davecan** [comment]
Wow. I was actually just going into my e-mail to respond to Derek that I have just forged ahead with plain CSS and would re-evaluate this at a later time, and lo and behold there's a fix! You guys rock!

**Comment by davecan** [comment]
Argh. Still doesn't work. :(

I downloaded the patched compass-install-rails.rb file and ran the command as you specified to create a new project to test it out. Below is the session I ran with the error output:

```
      What CSS Framework do you want to use with Compass? (default: 'blueprint')
960
      Where would you like to keep your sass files within your project? (default: 'app/stylesheets')

      Where would you like Compass to store your compiled css files? (default: 'public/stylesheets/compiled')

     gem  haml
     gem  chriseppstein-compass
    rake  gems:install GEM=haml
    D:/Dev/Languages/ruby/lib/ruby/gems/1.8/gems/activesupport-2.3.4/lib/active_support/core_ext/kernel/agnostics.rb:7:in ``': Exec format error - rake gems:install GEM=haml RAILS_ENV=development
```

I also still don't have the app/stylesheets directory. I'm not really sure what I'm doing wrong. Any ideas? :(

Thanks for the help.

*\* EDIT *\* Well of course I don't have the stylesheets directory. After looking at the code it never makes it that far, since it is dying on the HAML gem install. The specific error seems to be with how ActiveSupport is implementing the ` method to allow shell execution. I could manually execute each one of these commands, but I tried that last time and seemed to run into problems. Maybe I'll give it another try...

*\* EDIT 2 *\* OK I _can_ install it manually by piecing together the remaining commands from compass-install-rails.rb (i.e. manually running the rake commands, building the compass/init.rb file, etc) and it seems to work just fine. I now have SASS and HAML running and compass seems to be doing its thing. I just can't seem to figure out why it won't run automatically. I'm feeling pretty confident that this is NOT a bug in ActiveSupport, but beyond that I'm stumped... :(

**Comment by davecan** [comment]
Thanks Jonathan, that did work for me. That's essentially what I did yesterday and got a new project up and running, though I went a bit further and manually built the compass command from the compass-install-rails.rb file and also unpacked the gems into the vendor/gems directory. I then repeated the steps into an active project I'm working on and it worked fine. 

It was way too late by then to try and learn compass though so I stuck with CSS, but its nice knowing the simple approach so when I come back I can start it up easily and try it again. It looks like a really awesome little framework. Thanks for all the help, and hopefully through this someone else is helped by the easy install info you provided.

Thanks again.
-dave
